hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/protocol/EndpointService.java

```diff
@@ -443,21 +445,45 @@ public void measureMetrics(Collection<MeasurementInstance<L, MetricType<L>>> ins
             for (MeasurementInstance<L, MetricType<L>> instance : instances) {
                 AttributeLocation<L> location = instance.getAttributeLocation();
                 Object o = driver.fetchAttribute(location);
-                double value = 0;
-                if (o instanceof List<?>) {
-                    /* aggregate */
-                    List<?> list = (List<?>) o;
-                    for (Object item : list) {
-                        double num = toDouble(item);
-                        value += num;
+                Object metricValue; // will be either a String or Double
+                if (instance.getType().getMetricType() == org.hawkular.metrics.client.common.MetricType.STRING) {
+                    StringBuilder svalue = new StringBuilder();
+                    if (o instanceof List<?>) {
+                        /* aggregate */
+                        List<?> list = (List<?>) o;
+                        for (Object item : list) {
+                            svalue.append(String.valueOf(item));
```

Comment:
This is sort of weird, I guess we're just trying to do something reasonable with a list of Strings.  Perhaps CSV would be a better option than just stringing it all together with no delims.

Also, would be a little more efficient to just use the StringBuilder for the List case.